packages/cozy-stack-client/src/mangoIndex.js

```diff
@@ -80,14 +82,18 @@ export const transformSort = sort => {
  * Compute fields that should be indexed for a mango
  * query to work
  *
- * @param {MangoQueryOptions} options - Mango query options
- * @param {MangoSelector|null} options.selector - Mango selector
- * @param {MangoPartialFilter|null} options.partialFilter - An optional partial filter
+ * @param {object} options - Mango query options
+ * @param {MangoSelector} [options.selector] - Mango selector
+ * @param {MangoPartialFilter} [options.partialFilter] - An optional partial filter
  * @param {MangoSort} [options.sort] - The sorting parameters
  *
  * @returns {Array} - Fields to index
  */
-export const getIndexFields = ({ selector, partialFilter, sort = [] }) => {
+export const getIndexFields = ({
+  selector = [],
+  partialFilter = [],
```

Comment:
selector and partialFilter are not array :thinking:

Reply:
Thanks fixed. Always blown by `const a = []; typeof a` 

We should create a real type for selector with a more restrictive defintion (and so for partialFilter since partialFilter is a selector)